### PR TITLE
Spring Boot Data Jpa

### PR DESCRIPTION
**Pull request by aksoli666**
+replaced the existing BookRepository
with the one that will use JpaRepository interface +added Liquibase support
+changed spring.jpa.hibernate.ddl-auto property value to validate +implemented soft delete concept for the Book entity +added missing endpoints and methods for BookService +added endpoints:
PUT: /api/books/{id} (Update a specific book);
DELETE /api/books/{id} (Delete a specific book)

### CODE REVIEW

File: src/main/java/online/book/store/model/Book.java [hunk 1]
```diff
@@ -10,11 +10,15 @@
...
 import lombok.ToString;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Setter
 @Getter
 @ToString
 @Entity
+@SQLDelete(sql = "UPDATE employees SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
```
Comment (Rommelua) [inline src/main/java/online/book/store/model/Book.java:21]: UPDATE **employees**?

File: src/main/java/online/book/store/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -38,4 +39,24 @@ public List<BookDto> findAll() {
...
+    public BookDto updateById(Long id, UpdateBookRequestDto bookRequestDto) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can`t find book by id: " + id));
+
+        book.setTitle(bookRequestDto.getTitle());
+        book.setAuthor(bookRequestDto.getAuthor());
+        book.setIsbn("VbC1c" + new Random().nextInt());
+        book.setPrice(bookRequestDto.getPrice());
+        book.setDescription(bookRequestDto.getDescription());
+        book.setCoverImage(bookRequestDto.getCoverImage());
```
Comment (Rommelua) [inline src/main/java/online/book/store/service/impl/BookServiceImpl.java:53]: Check if ID exists and then use mapper to update the entity

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/online/book/store/model/Book.java [hunk 3]
```diff
@@ -10,11 +10,15 @@
...
 import lombok.ToString;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Setter
 @Getter
 @ToString
 @Entity
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
```
Comment (MAMentorFX) [inline src/main/java/online/book/store/model/Book.java:21]: @Where is deprecated, use @SQLRestriction

File: src/main/java/online/book/store/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -38,4 +39,17 @@ public List<BookDto> findAll() {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public BookDto updateById(Long id, UpdateBookRequestDto bookRequestDto) {
```
Comment (MAMentorFX) [inline src/main/java/online/book/store/service/impl/BookServiceImpl.java:44]:
```suggestion
    @Override
    @Transactional
    public BookDto updateById(Long id, UpdateBookRequestDto bookRequestDto) {
```

we make more than one call to db, so it has to be isolated

**Review by MAMentorFX — CHANGES_REQUESTED** [review]

File: src/main/java/online/book/store/model/Book.java [hunk 5]
```diff
@@ -10,11 +10,15 @@
...
 import lombok.ToString;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
 
 @Setter
 @Getter
 @ToString
 @Entity
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@SQLRestriction(value = "is_deleted=false")
```
Comment (okuzan) [inline src/main/java/online/book/store/model/Book.java:21]:
```suggestion
@SQLDelete(sql = "UPDATE books SET is_deleted = TRUE WHERE id = ?")
@SQLRestriction(value = "is_deleted = FALSE")
```

**Review by okuzan — APPROVED** [review]
good job

**Pull request merged**